VnodeCacheDesign.md

```diff
@@ -0,0 +1,101 @@
+# Vnode Cache Design
+
+## Goals
+
+- Reduce the time required to determine if a file is inside a virtualization root
+- Reduce the time required to find a file's virtualization root
+- Eliminate the use of file flags to determine if a file is inside of a root
+
+## Design 
+
+### Cached Data
+
+PrjFSKext will cache the following information for each `vnode_t` pointer:
+
+- The `vnode_t` itself
+- The vnode's `uint32_t` vid (vnode generation number)
+- A `uint32_t` virtualization\cache generation number (hereby referred to as `vrgid`).  If a cache entry's `vrgid` does not match its roots it means that all entries for that root have been invalidated and the `VirtualizationRootHandle` in the cache entry is no longer valid.  There will also be a global `vrgid`  for all vnodes that are not inside of any virtualization roots.
```

Comment:
I would perhaps not nail down the data type for this just yet; it looks like a uint32_t will push us over another 8-byte boundary, so instead of 4 bytes, it'll cost us 10 bytes per entry. A `uint16_t` might be good enough for this purpose, and would fit in the padding left by the rest of the fields.